### DISCUSSION TITLE
How to add a hidden custom attribute to BlockLayout

### DISCUSSION THREAD

**Discussion by hom** [mozman/ezdxf#726] [Q&A] [answered]
Hi, thanks much for this `ezdxf` tool, I want to add a custom **hidden** attribute to BlockLayout recently and use the attribute value to run some logic code. I try to use `add_attdef` and `add_attrib` function, But the attribute value always shown in AutoCAD, unless I set the attribute value is `""`.

I found the `ATTMODE` system variable in AutoCAD , if set this varible to `0`(default is `1`), the value will hide as I want. But I can not find a way to change this default `value` in `ezdxf`. And I found the variable is defined in `src/ezdxf/sections/headervars.py +180`, so what I should do to solve this problem?

Thanks very much, waiting for your reply

**Comment by mozman** [comment, marked as answer]
Setting header variables in the HEADER section:

```Python
doc.header["$ATTMODE"] = 0
```

Setting a block reference (`INSERT` entity) attribute invisible:

```Python
attribute = blockref.add_attribute("TAG_NAME", "VALUE_STR")
attribute.is_invisible = True
```

**Reply by hom** [reply]
Ok, thanks very much, got it and I will have a try : )
